model/adjuster/span_hash_deduper.go

```diff
@@ -19,15 +19,19 @@ func DedupeBySpanHash() Adjuster {
 
 type spanHashDeduper struct {
 	trace       *model.Trace
-	spansByHash map[uint64]*model.Span
+	spansByHash map[int64]*model.Span
 }
 
 func (d *spanHashDeduper) groupSpansByHash() {
-	spansByHash := make(map[uint64]*model.Span)
+	spansByHash := make(map[int64]*model.Span)
 	for _, span := range d.trace.Spans {
-		hash, _ := model.HashCode(span)
-		if _, ok := spansByHash[hash]; !ok {
-			spansByHash[hash] = span
+		spanHash, found := span.GetHashTag()
```

Comment:
is this approach going to work if the span undergoes multiple transformations? E.g. we start with sanitizers that can mutate it, and we can add the tag as the last step in sanitizers (so probably ok for the write path). But then on the read path we also have adjusters that can change the trace, with this deduper being somewhere in the middle, is it going to get consistent behavior? I suspect yes because as a deduper it's job is really to filter out identical spans _as ingested_.